pagination/src/main/java/com/cerner/beadledom/pagination/OffsetPaginatedList.java

```diff
@@ -93,14 +98,27 @@
      * Convenience method that creates metadata with total results and default field names
      * without having to manually build a {@link OffsetPaginatedListMetadata} object.
      *
-     * @param totalResults the total results available.
-     * @return The {@link Builder} being used for the {@link OffsetPaginatedList}.
+     * @param totalResults the total results available
+     * @return the {@link Builder} being used for the {@link OffsetPaginatedList}
+     * @deprecated use {@link #totalResults(Long)} instead
      */
+    @Deprecated
     public OffsetPaginatedList.Builder<T> metadata(@Nullable Long totalResults) {
+      return totalResults(totalResults);
+    }
+
+    /**
+     * Convenience method that adds a metadata instance with total results and default field names
+     * without having to manually build an {@link OffsetPaginatedListMetadata} object.
+     *
+     * @param totalResults the total results available
+     * @return the {@link Builder} being used for the {@link OffsetPaginatedList}
+     */
+    public OffsetPaginatedList.Builder<T> totalResults(@Nullable Long totalResults) {
       return metadata(
-          LimitParameter.getDefaultLimitFieldName(), null,
-          OffsetParameter.getDefaultOffsetFieldName(), null,
-          totalResults, null);
+          OffsetPaginatedListMetadata.builder()
```

Comment:
Possibly stupid question: Do we have to do argument validation for totalResults to ensure it doesn't have negative values?

Reply:
That's probably a good idea. Does anyone else have any passivity concerns of me adding such a check?

Reply:
No concerns from me.

Reply:
Autovalue recommends doing validation in the `build` method rather than directly in the setters [1]. This makes it a little bit less fail-fast, but should still be fine.

[1] https://github.com/google/auto/blob/master/value/userguide/builders-howto.md#validate

Reply:
https://github.com/cerner/beadledom/pull/146/commits/91732ff7059233b1f33f364e3e4b4409723a40ab